### PR TITLE
check for size of external medium before move-root-to-USB

### LINKED ISSUE #424: Move Root on Openhabian-config may need a check for memory stick size
I installed a 128mb memory stick by accident and used the move root.  The function worked fine, but the pi would not boot.  I suspect something happened at copy time or boot time as 128mb is too small.

I would request if possible maybe add a check to make sure memory stick is a minimum size.  Probably no less than the size of the installed SD card.  

Thanks!

### LINKED ISSUE #743: openHABian interactive install/menu option descriptions
Following up on a number of community posts, there's a number of options in openHABian that are unclear what they actually do and when you should or can use them.
Examples:
* 02 Upgrade System
It'll upgrade ALL packages on the system .... including openHAB2 !!
Which may be intentional or not - least thing is to make users aware of it.

* move root to USB
Does it work with any supported HW? Seems not out of the box, see e.g. #731 
How does it relate to ZRAM?
Situations to apply this in:
~~1) to move to an USB stick to avoid SD corruption=> bad idea as the USB is as-bad, better go ZRAM~~
~~2) to move to a "reliable" external medium such as SSD or HDD => ok, but don't use ZRAM on top~~

* ~~Delay rules load~~ --> fixed in #764

* ~~Mosquitto ? How does it relate to MQTTv2 ? Can only have one broker bind to the ports~~
--> there's discussions to deprecate moquette (OH-embedded broker) rather than to extend use
--> more people to switch back to use Mosquitto --> no real need to change openHABian docs

Need to think of the best way of documenting this to users per-issue.
Change menu text ? whiptail infobox ?

### PR DESCRIPTION
**Pull request by mstormi**
enhance in-program menu descriptions and infotexts on the move-root-to-USB option

Fixes: #424 
Fixes: #743 

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: functions/ext-storage.bash [hunk 1]
```diff
@@ -15,7 +15,7 @@ move_root2usb() {
 
   if [ -f /etc/ztab ]; then
     if [ -n "$INTERACTIVE" ]; then
-      whiptail --title "Incompatible selection detected" --msgbox "Move root to USB must not be used together with ZRAM.\\nIf you want to mitigate SD carsd corruption, don't move root but stay with ZRAM, it is the proper choice. If you want to move for other reasons, uninstall ZRAM first then return here." 10 60
+      whiptail --title "Incompatible selection detected" --msgbox "Move root to USB must not be used together with ZRAM.\\nIf you want to mitigate SD carsd corruption, don't move root but stay with ZRAM, it is the proper choice.\\nIf you want to move for other reasons, uninstall ZRAM first then return here." 10 65
```
Comment (radokristof) [inline functions/ext-storage.bash:18]: Typo: "carsd"

**Comment by mstormi** [comment]
or @ecdye?

File: functions/ext-storage.bash [hunk 2]
```diff
@@ -40,10 +40,25 @@ move_root2usb() {
   if ! [ $rootonsdcard = true ]; then
     infotext="It seems as if your system root is not on the SD card.
        ***Aborting, process cant be started***"
-    whiptail --title "System root not on SD card?" --msgbox "$infotext" 8 78
+    whiptail --title "System root not on SD card ?" --msgbox "$infotext" 8 78
```
Comment (ecdye) [inline functions/ext-storage.bash:43]: Why the extra space before the "?"?

File: functions/ext-storage.bash [hunk 3]
```diff
@@ -40,10 +40,25 @@ move_root2usb() {
...
     return 0
   fi
 
+  #exit if destination is not available
+  if ! [[ -b "$NEWROOTPART" ]]; then
+    infotext="Seems there is no external SD card inserted that we could move openHABian root to.\\n***Aborting, process cant be started***"
```
Comment (ecdye) [inline functions/ext-storage.bash:49]: "external SD card" would be better said as "external storage medium"

File: functions/ext-storage.bash [hunk 4]
```diff
@@ -40,10 +40,25 @@ move_root2usb() {
...
+    whiptail --title "No destination SD card ?" --msgbox "$infotext" 8 78
```
Comment (ecdye) [inline functions/ext-storage.bash:50]: extra space again

File: functions/ext-storage.bash [hunk 5]
```diff
@@ -40,10 +40,25 @@ move_root2usb() {
...
+    return 0
+  fi
+
+  srcsize=$(blockdev --getsize64 /dev/mmcblk0)
+  destsize=$(blockdev --getsize64 "$NEWROOTDEV")
+  if [[ "$srcsize" -gt "$destsize" ]]; then
+    infotext="Your internal SD card is larger than the external device you want to move your root to. This will very likely break your system.\\n\\nDo you still REALLY want to continue ?"
```
Comment (ecdye) [inline functions/ext-storage.bash:57]: here to.

File: functions/ext-storage.bash (hunk 2 above)
Comment (mstormi) [reply functions/ext-storage.bash:43]: my personal preference

File: functions/ext-storage.bash (hunk 4 above)
Comment (mstormi) [reply functions/ext-storage.bash:50]: see up

File: functions/ext-storage.bash (hunk 3 above)
Comment (mstormi) [reply functions/ext-storage.bash:49]: done

File: functions/ext-storage.bash (hunk 5 above)
Comment (mstormi) [reply functions/ext-storage.bash:57]: done

**Pull request merged**